src/components/shared/form_fields/useFieldModel.ts

```diff
@@ -0,0 +1,11 @@
+import { Ref, ref, UnwrapRef, watch } from 'vue';
+
+export function useFieldModel<T>( modelValue: () => UnwrapRef<T>, initialValue: T ): Ref<UnwrapRef<T>> {
```

Comment:
When you pass the model value straight to an input model from the props you get:

`VueCompilerError: v-model cannot be used on a prop, because local prop bindings are not writable.
Use a v-bind binding combined with a v-on listener that emits update:x event instead.`

Reply:
>Also, why is modelValue an anonymous function that returns something?

Looking at it again, I can see I ran into loads of TS typing issues when I try to pass a prop straight through